project/app/api/visualization.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+from fastapi import APIRouter
+
+from app.utils.security.header_checking import AuthRouteHandler
+from app.api.models import HistogramRequest, LineGraphRequest
+from app.utils.visualizations import histogram, line_graph
+
+# global variables and services
+router = APIRouter(route_class=AuthRouteHandler)
+log = logging.getLogger(__name__)
+
+
+@router.route("/viz/linegraph")
+def return_line_graph(data: LineGraphRequest):
+    """Endpoint produces a line graph of a student's squad scores over time
+
+        Input: A list containing the history of the students scores, in
+        chronological order. And student's name.
```

Comment:
Perhaps reword to shorten: "A list containing the student's name and history of scores in chronological order."

Reply:
The list containing the history of the student's scores and the student's name are two different inputs, not one list. But maybe edit this so that point is more clear to anyone else reading this.